src/Router/PreloadLink.tsx

```diff
@@ -1,177 +1,281 @@
 import * as Found from "found"
-import { isEmpty, isFunction, isUndefined, pick } from "lodash/fp"
+import { withRouter } from "found"
+import { Link } from "found"
+import { compose, isEmpty, isUndefined, last, pick } from "lodash/fp"
 import PropTypes from "prop-types"
-import React, { Component } from "react"
+import React from "react"
 import { fetchQuery } from "react-relay"
+import { QueryRendererProps } from "react-relay"
+import { Container, Subscribe } from "unstated"
 import { ContextConsumer } from "../Components/Artsy"
 import { PreloadLinkProps, PreloadLinkState } from "./types"
-export { Link } from "found"
-
-export const PreloadLink = Found.withRouter<PreloadLinkProps>(props => {
-  const PreloadLinkWrapper = ContextConsumer(
-    class extends Component<PreloadLinkProps, PreloadLinkState> {
-      static propTypes = {
-        Component: PropTypes.any,
-        immediate: PropTypes.bool, // load route data transparently in the bg
-        onToggleLoading: PropTypes.func,
-        relayEnvironment: PropTypes.object.isRequired,
-        reactionRouter: PropTypes.shape({
-          routes: PropTypes.array.isRequired,
-          resolver: PropTypes.object.isRequired,
-        }).isRequired,
-        replace: PropTypes.string,
-        to: PropTypes.string,
-        name: PropTypes.string,
-      }
 
-      static defaultProps = {
-        immediate: false,
-        onToggleLoading: x => x,
-      }
+/**
+ * Preload link state for use with unstated. If needing to tap into loading
+ * status can use like so:
+ *
+ * @example
+ *
+ * return(
+ *   <Subscribe to={[State]}>
+ *     {({ isFetching }) => {
+ *       return (
+ *         <PreloadLink to='/some-path'>
+ *           Fetching: {isFetching}
+ *         </PreloadLink>
+ *       )
+ *     }}
+ *   </Subscribe>
+ * )
+ */
+export class State extends Container<PreloadLinkState> {
```

Comment:
This export pattern lets us take state that we're interested and "subscribe" to it anywhere in the tree. Keeps things nice and decoupled and great for lib code.

Reply:
I feel like I’m missing something, the example being used in this change (and that I commented about below) is probably too contrived for me to see the bigger picture.